### ISSUE TITLE
`undefined < 1` results in true

### ISSUE THREAD

**Issue by sarahgp** [tessel/runtime#704]
Contrary to other js runtimes (and possibly the standard?), `undefined < 1` returns `true`.

**Comment by natevw** [comment]
This is probably directly related to https://github.com/tessel/runtime/issues/187. In JS you have:

```
assert.equal(undefined < 1, false);
assert.equal(null < 1, true);
```

But under Tessel's runtime there is no difference between `null` and `undefined`. It appears that in your case the internal `nil` value is being treated as if it were `null` value rather than `undefined`.

**Comment by rwaldron** [comment]
> But under Tessel's runtime there is no difference between null and undefined. 

Yes, courtesy of Lua. There are other issues with this as well, but the most egregious: 

```js
console.log(undefined === null); // true
```

Which is broken because

1. `undefined === null; // false` because Strict Equality Comparison _does not_ coerce
2. `undefined == null; // true` because Abstract Equality Comparison _does_ coerce
3. Since the tessel runtime breaks the semantic invariants defined in 1 & 2, it's a hostile environment—not just because they shouldn't equal, but because the runtime can't be trusted to evaluate a piece of code correctly.

Take a look at this example: 

```js
[undefined, null, , undefined].forEach(function(value, index) {
  console.log(index, typeof value);
});
```

Thats: "undefined", "null", an elision (or hole), "undefined"

Here's the correct output:

```
0 'undefined'
1 'object'
3 'undefined'
```

Note the skipped elision/hole—that's correct.

Here's tessel runtime output: 

```
0 undefined
1 undefined
2 undefined
3 undefined
```

1. Did not skip hole
2. cannot tell difference between null, undefined or elision
3. typeof thinks null is undefined
